codalab/lib/bundle_cli.py

```diff
@@ -2335,24 +2331,84 @@ def do_events_command(self, args):
             Commands.Argument('-t', '--time-quota', help='Total amount of time allowed (e.g., 3, 3m, 3h, 3d)'),
             Commands.Argument('-d', '--disk-quota', help='Total amount of disk allowed (e.g., 3, 3k, 3m, 3g, 3t)'),
+            Commands.Argument('--first-name', help='First name'),
+            Commands.Argument('--last-name', help='Last name'),
+            Commands.Argument('--affiliation', help='Affiliation'),
+            Commands.Argument('--url', help='Website URL'),
         ),
     )
     def do_uedit_command(self, args):
         """
         Edit properties of users.
         """
-        client = self.manager.current_client()
-        user_info = client.get_user_info(args.user_id)
-        if args.time_quota is not None:
-            user_info['time_quota'] = formatting.parse_duration(args.time_quota)
-        if args.disk_quota is not None:
-            user_info['disk_quota'] = formatting.parse_size(args.disk_quota)
-        client.update_user_info(user_info)
+        client = self.manager.current_client(use_rest=True)
+
+        # Build user info
+        user_info = {}
+        for key in ('time_quota', 'disk_quota', 'first_name', 'last_name',
+                    'affiliation', 'url'):
+            if getattr(args, key) is not None:
+                user_info[key] = getattr(args, key)
+
+        if not user_info:
+            raise UsageError("No fields to update.")
+
+        # If user id is not specified, update the authenticated user
+        if args.user_spec is None:
+            user = client.update_authenticated_user(user_info)
+            self.print_user_info(user, include_private=True)
+        else:
+            user_info['id'] = client.fetch('users', args.user_spec)['id']
+            user = client.update('users', [user_info])[0]
+            self.print_user_info(user, include_private=False)
+
+    @Commands.command(
+        'uinfo',
+        help=[
+            'Show user information.',
+        ],
+        arguments=(
+            Commands.Argument('user_spec', nargs='?', help='Username or id of user to show [default: the authenticated user]'),
+        ),
+    )
+    def do_uinfo_command(self, args):
+        """
+        Edit properties of users.
+        """
+        client = self.manager.current_client(use_rest=True)
+
+        if args.user_spec is None:
+            user = client.fetch('user')
+            self.print_user_info(user, include_private=True)
+        else:
+            user = client.fetch('users', args.user_spec)
+            self.print_user_info(user, include_private=False)
+
+    def print_user_info(self, user, include_private=False):
+        for key in ('id', 'user_name', 'first_name', 'last_name',
+                    'affiliation', 'url', 'date_joined'):
+            print >>self.stdout, u'{:<15}: {}'.format(key, user.get(key, None))
+
+        if include_private:
```

Comment:
Put comment that if we're not user or root, we won't get these fields back, so it's clear that we're not relying on the client to do privacy.